pkg/storage/allocator.go

```diff
@@ -541,7 +541,8 @@ func filterBehindReplicas(
 	candidates := make([]roachpb.ReplicaDescriptor, 0, len(replicas))
 	for _, r := range replicas {
 		if progress, ok := raftStatus.Progress[uint64(r.ReplicaID)]; ok {
-			if progress.State == raft.ProgressStateReplicate &&
+			if (progress.State == raft.ProgressStateReplicate ||
+				progress.State == raft.ProgressStateProbe) &&
```

Comment:
While the leader is in `ProgressStateProbe`, I think that is accidental on the part of the Raft code. I'd rather have an explicit check for the leader replica: `uint64(r.ReplicaID) == raftStatus.Lead`.